File: sorting/selection_sort.cpp
// Problem
//https://www.hackerearth.com/practice/algorithms/sorting/selection-sort/practice-problems/algorithm/selection-sort-problem/?utm_source=header&utm_medium=search&utm_campaign=he-search

#include <bits/stdc++.h>
using namespace std;

int main() {
	long n,x;
	cin >> n>>x;
	int ar[n];
	for(int i=0;i<n;i++)
		cin>>ar[i];
	int min,index=0;
	for(int i=0;i<x && i<n-1;i++){
		min = i;
		for(int j=i+1;j<n;j++){
			if(ar[min] > ar[j])	{
				min = j;
			}
		}
		int tmp = ar[i];
		ar[i] = ar[min];
		ar[min] = tmp;
	}
	for(int i=0;i<n;i++)
		cout<<ar[i]<<" ";
}

File: sorting/selection_sort.js
var input = [1,5,4,2,3];
var m = new Map();
for(var i=0;i<input.length;i++){
    m.set(input[i],i);
}
console.log(m);
insertion_sort(input,input.length);
function insertion_sort(input,n) {
    for (let i = 0; i  < n; i++) {
        var tmp = input[i];
        var j=i;
        while(j>0 && input[j-1] > tmp){
            input[j] = input[j-1];
            j--;
        }
        input[j] = tmp;
    }
}